### PR TITLE
New Router Gateway Cheat Sheet

### PR DESCRIPTION
**Pull request by benabrahamson**
This cheat sheet contains a list of links for popular router brands' gateway IP address.

---

Instant Answer Page: https://duck.co/ia/view/router_gateway_reference

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by benabrahamson** [comment]
Hmmm... looks like it doesn't like the IA page I made. What can I do to solve this?

File: share/goodie/cheat_sheets/json/routergateway.json [hunk 1]
```diff
@@ -0,0 +1,1009 @@
+{
+    "id": "router_gateway_reference",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/routergateway.json]: As noted in Travis build log

File name must end with _cheat_sheet

Comment (kirkins) [inline share/goodie/cheat_sheets/json/routergateway.json]: **Correction** ID must be file name + _cheat_sheet

So in this case your json file will be called: `router-gateway-reference`

The ID will be: `router_gateway_reference_cheat_sheet`

We try to keep the style the same across all cheat sheets

If in doubt reference the cheat sheet folder https://github.com/duckduckgo/zeroclickinfo-goodies/tree/master/share/goodie/cheat_sheets/json

**Comment by kirkins** [comment]
@benabrahamson also could run the json file through this beautifier with indent set to 4.

**Comment by benabrahamson** [comment]
Hi @kirkins, thanks for your help. I have successfully beautified the file, as you said, however I am still having issues with the ID. 

According to the Travis Build log
`Cheat sheet with ID "router_gateway_reference_cheat_sheet" not found in metadata.`

I think the issue might lie with the IA page itself not having the right ID.

(In addition, optimally I think that maybe the actual name of the cheat sheet should drop the "reference" part all together - it is not a necessary part of the name.)

I seem unable to edit the IA page to fix the renaming and ID issue. How should I go about doing that?

Thanks again!

File: share/goodie/cheat_sheets/json/router-gateway-reference.json [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+        "Planet",
+        "Repotec",
+        "Senao",
+        "Siemens",
+        "Sitecom",
+        "SMC Networks",
+        "Sonicwall",
+        "SpeedTouch",
+        "Sweex",
+        "Thomson",
```
Comment (kirkins) [inline share/goodie/cheat_sheets/json/router-gateway-reference.json]: You have a Tenda section but it's not in the section order array. Can you add it above Thomson.

`Failed test 'FAIL: The following sections were used in 'sections' but not defined in 'section_order:' ('Tenda')'`

**Comment by kirkins** [comment]
@benabrahamson I will get back to you on the ID issue if someone else doesn't chime in first.

**Comment by benabrahamson** [comment]
@kirkins I've added "Tenda" to to section order. Must have missed that one! Thanks :+1:

**Comment by kirkins** [comment]
@benabrahamson great it looks like it's passing the tests now. I'm just here to help people get their cheat sheets into the expected format. 

Someone from duckduckgo will be able to give you more feedback or accept it as is and merge it.

**Comment by mintsoft** [comment]
You could add `Fortigate`; they're usually on `192.168.1.99` or `192.168.1.2`

**Comment by benabrahamson** [comment]
@sahildua2305 I have actually already made the IP addresses into links :+1:

**Comment by benabrahamson** [comment]
@mintsoft Thanks, have added that suggestion. I have also added a "Generic" section for routers that are not listed.

**Comment by sahildua2305** [comment]
@benabrahamson Sorry, I wasn't clear probably. I meant it doesn't really make sense to have those IPs as links. Why do you think the otherwise?

**Comment by mintsoft** [comment]
@benabrahamson ping! Just checking in, are you OK with the feedback?

**Comment by benabrahamson** [comment]
@sahildua2305 My reasoning behind having the IPs as links was to make it easier to access the router gateway from the browser. i.e if someone wanted to change the SSID of their network, for example, they could click on the link to access the router config page, and do that, or whatever.

**Comment by sahildua2305** [comment]
@benabrahamson May be I am understanding it wrong. Please correct me if that's the case.

According to my understanding, people will need to copy these router gateway IPs if they want to use any of these. So making them as links doesn't make sense to me. Plain text will be better. What do you think about this?

**Comment by benabrahamson** [comment]
@sahildua2305 People would need to copy them if they wanted to use the IP in the command line or something like that, but if they wanted to access their router config page, they could just click the link. I'm talking about the config page accessed via your browser.

**Comment by sahildua2305** [comment]
Okay, looks good then. Thanks for explaining @benabrahamson 😄

**Pull request merged**